pulsar-discovery-service/src/main/java/org/apache/pulsar/discovery/service/server/DiscoveryServiceStarter.java

```diff
@@ -51,13 +51,17 @@ public static void init(String configFile) throws Exception {
         // load config file
         final ServiceConfig config = PulsarConfigurationLoader.create(configFile, ServiceConfig.class);
         checkArgument(!isEmpty(config.getZookeeperServers()), "zookeeperServers must be provided");
-        checkArgument(!isEmpty(config.getGlobalZookeeperServers()), "global-zookeeperServers must be provided");
-        
+        if(!isEmpty(config.getGlobalZookeeperServers())) {
```

Comment:
I think the logic here is not that good.

it is probably better to check if 'configuaration-store servers' is empty or not. if it is not empty, it is okay to have an empty global zookeeper servers, otherwise checkout global zookeeper server is empty or not. you only setConfigurationStoreServer when configurationStoreServer is empty.

Reply:
A unit test for testing this behavior seems to be required. Otherwise it is a bit hard to know whether we break BC or not.